specs/02-data-availability-sampling.md

```diff
@@ -0,0 +1,242 @@
+# Data Availability Sampling
+
+Child shards are propagating to the upper layers of the system commitments of their blocks through
+`BlockInfo`, and segment commitments to the beacon chain through `SegmentInfo` transactions.
+However, in order to be able to verify these blocks and segments, and to ensure that segments are
+always available for plotting, they all must be available for retrieval.
+
+This is done through a data availability sampling protocol that allows any node in the system to
+verify that blocks and segments are available for retrieval. The protocol is based on the idea of
+sampling a small number of chunks from encoded segments and blocks and verifying that they are
+available.
+
+### Availability sampling.
+
+This protocol is used to sample both: blocks and segments in shards.
+
+#### 1. Sampling process
+
+The sampling process is the same for both, blocks and segments. Any node in the system, but in
+particular light nodes, periodically sample random blocks and segments from different shards. We
+expect nodes of any kind to prioritise sampling the shards that they are interested in, but they are
+all free to choose where they sample.
+
+The sampling process is as follows:
+
+1.  **Select a Chunk**:
+    - Choose a random share index `i` from `0` to `total_shares - 1`.
+    - Choose a random chunk index `j` from `0` to `num_chunks_per_share - 1`.
+2.  **Request Data**:
+    - Request chunk `j` from share `i` and its witness `π_{i,j}` from a node (e.g., farmer).
```

Comment:
I would try to not confuse node and farmer.

I wrote this back in 2022 (with some updates since) that tries to clarify what different terms mean: https://subspacelabs.notion.site/Proposed-consistent-naming-for-many-things-in-Subspace-362312a436d640ca9fae262d8d1b1b14?pvs=74

It is slightly outdated when it comes to things like "Subspace Desktop", "Raw Record" is no longer present in this fork, decouples execution/operators are are not a thing here either, but overall it is still a useful vocabulary to agree on (I'll probably have to fork it here).